shared/src/utils/validators/is-value.ts

```diff
@@ -0,0 +1,25 @@
+import type { ValidationOptions, ValidationArguments } from 'class-validator';
+import { isIn } from 'class-validator';
+import type { GenericPropertyDecorator } from './generic-property-decorator';
+import { makeValidator } from './make-validator';
+
+export function isValue<T>(
+    expectedValue: T,
+    valueToBeValidated: unknown
+): boolean {
+    return isIn(valueToBeValidated, [expectedValue]);
+}
+
+// With TS 5.0 use `const T` (https://github.com/microsoft/TypeScript/pull/51865)
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function IsValue<T, Each extends boolean = false>(
```

Comment:
Does this work with arrays and object literals like `{ a: 0 }`?

Please add a class comment for this here. (compares shallow/deep by value/reference wether the decorated property has the specified value)

Reply:
I'll add a comment stating that it's using `class-validator`'s `isIn` which in turn currently uses `===`.